### ISSUE TITLE
Integration not working in HA 2024.02

### ISSUE THREAD

**Issue by kbabik** [Antoni-Czaplicki/vulcan-for-hassio#50]
Hi, After update HA to version 2024.02.01 integration not starting 
integration: v0.15.1
log:

`Logger: homeassistant.setup
Source: setup.py:251
First occurred: 11:16:01 (1 occurrences)
Last logged: 11:16:01

Setup failed for custom integration 'vulcan': Unable to import component: No module named 'imp'
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/setup.py", line 251, in _async_setup_component
    component = integration.get_component()
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/loader.py", line 827, in get_component
    ComponentProtocol, importlib.import_module(self.pkg_path)
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/importlib/__init__.py", line 90, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "<frozen importlib._bootstrap>", line 1387, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1360, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1331, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 935, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 994, in exec_module
  File "<frozen importlib._bootstrap>", line 488, in _call_with_frames_removed
  File "/config/custom_components/vulcan/__init__.py", line 11, in <module>
    from vulcan import Account, Keystore, UnauthorizedCertificateException, Vulcan
  File "/usr/local/lib/python3.12/site-packages/vulcan/__init__.py", line 3, in <module>
    from ._account import Account
  File "/usr/local/lib/python3.12/site-packages/vulcan/_account.py", line 3, in <module>
    from related import IntegerField, StringField, immutable
  File "/usr/local/lib/python3.12/site-packages/related/__init__.py", line 16, in <module>
    from .fields import (
  File "/usr/local/lib/python3.12/site-packages/related/fields.py", line 3, in <module>
    from future.moves.urllib.parse import ParseResult
  File "/usr/local/lib/python3.12/site-packages/future/moves/__init__.py", line 5, in <module>
    from future.standard_library import import_top_level_modules
  File "/usr/local/lib/python3.12/site-packages/future/standard_library/__init__.py", line 65, in <module>
    import imp
ModuleNotFoundError: No module named 'imp'`

**Comment by Antoni-Czaplicki** [comment]
Simply update to v0.15.1

**Comment by kbabik** [comment]
> Simply update to v0.15.1

that's the first thing i done, but it's not working either

**Comment by Antoni-Czaplicki** [comment]
Are you sure? And have you restarted home assistant instance?  Cause this update should prepare the integration for python 3.12 and prevent that issue. Also you could try to redownload the integration from HACS

**Comment by tomekswx** [comment]
It doesn't work for me either, despite reinstallations and reboots, error log below:

2024-02-12 12:39:39.907 ERROR (MainThread) [homeassistant.config_entries] Error setting up entry *********** for vulcan
Traceback (most recent call last):
File "/usr/src/homeassistant/homeassistant/config_entries.py", line 444, in async_setup
result = await component.async_setup_entry(hass, self)
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
File "/config/custom_components/vulcan/__init__.py", line 28, in async_setup_entry
await client.select_student()
File "/usr/local/lib/python3.12/site-packages/vulcan/_client.py", line 42, in select_student
students = await self.get_students()
^^^^^^^^^^^^^^^^^^^^^^^^^
File "/usr/local/lib/python3.12/site-packages/vulcan/_client.py", line 61, in get_students
self._students = await Student.get(self._api)
^^^^^^^^^^^^^^^^^^^^^^^^^^^^
File "/usr/local/lib/python3.12/site-packages/vulcan/model/_student.py", line 79, in get
return [Student.load(student) for student in data]
^^^^^^^^^^^^^^^^^^^^^
File "/usr/local/lib/python3.12/site-packages/vulcan/model/_serializable.py", line 49, in load
return to_model(cls, data)
^^^^^^^^^^^^^^^^^^^
File "/usr/local/lib/python3.12/site-packages/related/functions.py", line 104, in to_model
value = cls(**value)
^^^^^^^^^^^^
File "<attrs generated init vulcan.model._student.Student>", line 11, in __init__
__attr_validator_class_(self, __attr_class_, self.class_)
File "/usr/local/lib/python3.12/site-packages/attr/validators.py", line 106, in __call__
raise TypeError(
TypeError: ("'class_' must be (<class 'str'>,) (got None that is a <class 'NoneType'>).", Attribute(name='class_', default=NOTHING, validator=<instance_of validator for type (<class 'str'>,)>, repr=True, eq=True, eq_key=None, order=True, order_key=None, hash=None, init=True, metadata=mappingproxy({'key': 'ClassDisplay'}), type=<class 'str'>, converter=<function str_if_not_none at 0x7f12484ce0c0>, kw_only=False, inherited=False, on_setattr=None, alias='class_'), (<class 'str'>,), None)

**Comment by Antoni-Czaplicki** [comment]
@tomekswx the errors that you provided suggests a separate problem, please open another issue for it.

**Comment by kbabik** [comment]
> Are you sure? And have you restarted home assistant instance? Cause this update should prepare the integration for python 3.12 and prevent that issue. Also you could try to redownload the integration from HACS

yep, i'm pretty sure that i done everything right here, and got v0.15.1 installed, HA restarted etc... 
![image](https://github.com/Antoni-Czaplicki/vulcan-for-hassio/assets/103383354/b934019e-d2f4-4d40-914c-bc3a6cad55d9)

**Comment by ruszok** [comment]
Same here.0.15.1, HA 2024.2.1 and same error. Everything unavailable.

**Comment by Antoni-Czaplicki** [comment]
@ruszok @kbabik thanks for the info, I will try to find a fix but I can't promise when it will be available since I'm on a trip to Sahara desert right now 🌵. If you want to try to fix it on your own I think it has something to do with replacing in the api library the outdated package "related" with "related-without-future" that supports python 3.12 but also has the same folder name as the previous one. Probably it's some dependency issue that causes the integration to still use the old one

**Comment by ruszok** [comment]
Ok. We will see. Thanks!

Have nice trip.

**Comment by kbabik** [comment]
I just update HA to version 2024.2.2 and in this version vulkan integration works fine

**Comment by ruszok** [comment]
Same here - confirmed 2024.2.2 works fine again

**Comment by formiification** [comment]
Same story here, but also can't get in run on 2024.2.2. Can't even remove this integration as it still requires 'imp' module to do that. Once I click remove on Vulcan it creates log entry:
`Logger: aiohttp.server
Source: /usr/local/lib/python3.12/site-packages/aiohttp/web_protocol.py:421
First occurred: 09:53:29 (1 occurrences)
Last logged: 09:53:29

Error handling request
Traceback (most recent call last):
  File "/usr/local/lib/python3.12/site-packages/aiohttp/web_protocol.py", line 452, in _handle_request
    resp = await request_handler(request)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/aiohttp/web_app.py", line 543, in _handle
    resp = await handler(request)
           ^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/aiohttp/web_middlewares.py", line 114, in impl
    return await handler(request)
           ^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/components/http/security_filter.py", line 91, in security_filter_middleware
    return await handler(request)
           ^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/components/http/forwarded.py", line 227, in forwarded_middleware
    return await handler(request)
           ^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/components/http/request_context.py", line 28, in request_context_middleware
    return await handler(request)
           ^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/components/http/ban.py", line 80, in ban_middleware
    return await handler(request)
           ^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/components/http/auth.py", line 235, in auth_middleware
    return await handler(request)
           ^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/components/http/headers.py", line 31, in headers_middleware
    response = await handler(request)
               ^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/components/http/view.py", line 149, in handle
    result = await handler(request, **request.match_info)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/components/config/config_entries.py", line 88, in delete
    result = await hass.config_entries.async_remove(entry_id)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/config_entries.py", line 1280, in async_remove
    await entry.async_remove(self.hass)
  File "/usr/src/homeassistant/homeassistant/config_entries.py", line 641, in async_remove
    component = integration.get_component()
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/loader.py", line 827, in get_component
    ComponentProtocol, importlib.import_module(self.pkg_path)
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/importlib/__init__.py", line 90, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "<frozen importlib._bootstrap>", line 1387, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1360, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1331, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 935, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 994, in exec_module
  File "<frozen importlib._bootstrap>", line 488, in _call_with_frames_removed
  File "/config/custom_components/vulcan/__init__.py", line 11, in <module>
    from vulcan import Account, Keystore, UnauthorizedCertificateException, Vulcan
  File "/usr/local/lib/python3.12/site-packages/vulcan/__init__.py", line 3, in <module>
    from ._account import Account
  File "/usr/local/lib/python3.12/site-packages/vulcan/_account.py", line 3, in <module>
    from related import IntegerField, StringField, immutable
  File "/usr/local/lib/python3.12/site-packages/related/__init__.py", line 16, in <module>
    from .fields import (
  File "/usr/local/lib/python3.12/site-packages/related/fields.py", line 3, in <module>
    from future.moves.urllib.parse import ParseResult
  File "/usr/local/lib/python3.12/site-packages/future/moves/__init__.py", line 5, in <module>
    from future.standard_library import import_top_level_modules
  File "/usr/local/lib/python3.12/site-packages/future/standard_library/__init__.py", line 65, in <module>
    import imp
ModuleNotFoundError: No module named 'imp'
`

**Comment by Antoni-Czaplicki** [comment]
Hi @formiification, just to confirm - does the issue still presist?

**Comment by Antoni-Czaplicki** [comment]
For everyone having this issue:

1. Find integrations `__init__.py` file (located in your config directory and next inside `custom_components/vulcan/`
2. Edit this file and put the code attached below at the begging of the file
```
import subprocess

subprocess.run(["pip", "uninstall", "related", "-y"])
subprocess.run(["pip", "install", "related-without-future==0.7.4", "--force-reinstall"])
```
3. Restart Home Assistant
4. Delete that code from `__init__.py` file (You can also redownload the integration from HACS)

**Comment by formiification** [comment]
> Hi @formiification, just to confirm - does the issue still presist?

No, fixed once I've upgraded to 2024.2.5. Thanks for your support
